File: OneCodeTeam/Get-MAC-of-local-Ethernet-and-Wireless80211-17238/[VB]-Get MAC of local Ethernet and Wireless80211 (VBMACAddress)/VB/VBMACAddress/WMIHelper/WMIPropertyAttribute.vb
'************************** Module Header ******************************'
' Module Name:  WMIPropertyAttribute.vb
' Project:      VBMACAddress
' Copyright (c) Microsoft Corporation.
'
' This attribute indicates that the property is related to a WMI object 
' property.
' 
' 
' 
' This source is subject to the Microsoft Public License.
' See http://www.microsoft.com/opensource/licenses.mspx#Ms-PL.
' All other rights reserved.
' 
' THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND, 
' EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED 
' WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
'*************************************************************************'


Namespace WMIHelper
    <AttributeUsage(AttributeTargets.Property)>
    Public Class WMIPropertyAttribute
        Inherits Attribute

        ''' <summary>
        ''' The related WMI object property name.
        ''' </summary>
        Public Property PropertyName() As String

        ''' <summary>
        ''' The type of the related WMI object property value.
        ''' </summary>
        Public Property PropertyType() As WMIPropertyType

        ''' <summary>
        ''' Specify the .NET type if the WMI object property is referring to another 
        ''' WMI object.
        ''' </summary>
        Public Property AssociatedWMIClass() As Type
    End Class

End Namespace

File: OneCodeTeam/Get-MAC-of-local-Ethernet-and-Wireless80211-17237/[C#]-Get MAC of local Ethernet and Wireless80211 (CSMACAddress)/C#/CSMACAddress/WMIHelper/WMIPropertyAttribute.cs
/****************************** Module Header ******************************\
 * Module Name:  WMIPropertyAttribute.cs
 * Project:      CSMACAddress
 * Copyright (c) Microsoft Corporation.
 *
 * This attribute indicates that the property is related to a WMI object 
 * property.
 * 
 * 
 * 
 * This source is subject to the Microsoft Public License.
 * See http://www.microsoft.com/opensource/licenses.mspx#Ms-PL.
 * All other rights reserved.
 * 
 * THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND, 
 * EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED 
 * WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
\***************************************************************************/

using System;

namespace CSMACAddress.WMIHelper
{
    [AttributeUsage(AttributeTargets.Property)]
    public class WMIPropertyAttribute : Attribute
    {

        /// <summary>
        /// The related WMI object property name.
        /// </summary>
        public string PropertyName { get; set; }

        /// <summary>
        /// The type of the related WMI object property value.
        /// </summary>
        public WMIPropertyType PropertyType { get; set; }

        /// <summary>
        /// Specify the .NET type if the WMI object property is referring to another 
        /// WMI object.
        /// </summary>
        public Type AssociatedWMIClass { get; set; }
    }

}
